src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -112,10 +113,35 @@ ErrorIfRelationIsAKnownShard(Oid relationId)
 	}
 
 	const char *relationName = get_rel_name(relationId);
+
 	ereport(ERROR, (errmsg("relation \"%s\" is a shard relation ", relationName)));
 }
 
 
+/*
+ * ErrorIfChangingKnownShard errors out if the relation with relationId is
+ * a known shard and manual changes on known shards are disabled. This is
+ * valid for only non-citus (external) connections.
+ */
+void
+ErrorIfChangingKnownShard(Oid relationId)
+{
+	if (IsCitusInitiatedRemoteBackend() || EnableManualChangesToShards)
+	{
+		return;
+	}
+
+	/* search the relation in all schemas */
+	bool onlySearchPath = false;
+	if (RelationIsAKnownShard(relationId, onlySearchPath))
+	{
+		const char *relationName = get_rel_name(relationId);
+		ereport(ERROR, (errmsg("relation \"%s\" is a shard relation ", relationName),
+						(errhint("see GUC citus.enable_manual_changes_to_shards"))));
```

Comment:
Maybe: `To modify shards directly, set citus.enable_manual_changes_to_shards to on`

(that way you can copy-paste)